sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,1126 @@
+     * and show statistics.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} contains the
+     * {@link DocumentResultCollection batch} of {@link DetectLanguageResult detected languages}.
+     * @throws NullPointerException if {@code inputs} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<DocumentResultCollection<DetectLanguageResult>>> detectBatchLanguagesWithResponse(
+        List<DetectLanguageInput> inputs, TextAnalyticsRequestOptions options) {
+        try {
+            return withContext(
+                context -> detectBatchLanguagesWithResponse(inputs, options, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<DocumentResultCollection<DetectLanguageResult>>> detectBatchLanguagesWithResponse(
+        List<DetectLanguageInput> inputs, TextAnalyticsRequestOptions options, Context context) {
+        // TODO (savaity): validate inputs?
+        final LanguageBatchInput languageBatchInput = new LanguageBatchInput().setDocuments(inputs);
+        // TODO (savaity): confirm if options null is fine?
+        return service.languagesWithRestResponseAsync(
+            languageBatchInput, options == null ? null : options.getModelVersion(),
+            options == null ? null : options.showStatistics(), context)
+            .doOnSubscribe(ignoredValue -> logger.info("A batch of language input - {}", languageBatchInput))
+            .doOnSuccess(response -> logger.info("A batch of detected language output - {}", languageBatchInput))
+            .doOnError(error -> logger.warning("Failed to detected languages - {}", languageBatchInput))
+            .map(response -> new SimpleResponse<>(response, toDocumentResultCollection(response.getValue())));
+    }
+
+    /**
+     * Helper method to convert the service response of {@link LanguageResult} to {@link DocumentResultCollection}.
+     *
+     * @param languageResult the {@link LanguageResult} returned by the service.
+     * @return the {@link DocumentResultCollection} of {@link DetectLanguageResult} to be returned by the SDK.
+     */
+    private DocumentResultCollection<DetectLanguageResult> toDocumentResultCollection(
+        final LanguageResult languageResult) {
+        return new DocumentResultCollection<>(getDocumentLanguages(languageResult), languageResult.getModelVersion(),
+            languageResult.getStatistics());
+    }
+
+    /**
+     * Helper method to get a combined list of error documents and valid documents.
+     *
+     * @param languageResult the {@link LanguageResult} containing both the error and document list.
+     * @return the combined error and document list.
+     */
+    private static List<DetectLanguageResult> getDocumentLanguages(final LanguageResult languageResult) {
+        Stream<DetectLanguageResult> validDocumentList = languageResult.getDocuments().stream()
```

Comment:
convertToDetectLanguageResult and convertToErrorDetectLanguageResult are only used once, do we really need a function to do this? I think a simple foreach loop works.

Reply:
I was thinking to have one generic helper function for those two methods. But want to proof the end2end works first. I will reconsidier the approach with your suggestion as well.

Reply:
Methods are pieces of code that can be reused multiple times. Having a method used only once makes this code less readable because now I have to hop around. Also, when methods are invoked, it has a performance hit by adding another frame to the callstack. I don't think it's worth it for this piece.